### PR TITLE
Add UL2 (just docs)

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Adds docs for UL2: https://huggingface.co/google/ul2 -> important model that deserves its own doc page IMO

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/en/model_doc/ul2.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+Tips:
+
+- UL2 is an encoder-decoder model pre-trained on a mixture of denoising functions as well as fine-tuned on an array of downstream tasks.
+- UL2 has the same architecture as [T5v1.1](t5v1.1) but uses the Gated-SiLU activation function instead of Gated-GELU.
+- The authors release checkpoints of one architecture which can be seen [here](https://huggingface.co/google/ul2)
+
+The original code can be found [here](https://github.com/google-research/google-research/tree/master/ul2).
+
+This model was contributed by [DanielHesslow](https://huggingface.co/Seledorn).
```
Comment (patrickvonplaten) [inline docs/source/en/model_doc/ul2.mdx:31]: Make sure Daniel is the author

**Review by sgugger — APPROVED** [review]
Thanks for adding this! I'm curious why the empty module? Does one of the script complain if we don't add it?

**Comment by NielsRogge** [comment]
Shouldn't we also add a conversion script? Or isn't this required?

**Comment by patrickvonplaten** [comment]
> Shouldn't we also add a conversion script? Or isn't this required?

It should be the same as: https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/convert_t5x_checkpoint_to_flax.py 

cc @DanielHesslow

**Comment by DanielHesslow** [comment]
Yeah I based the conversion off of that script and mostly added a bunch of hacks to work around limitations of my local system. At some point there should probably be a stable conversion script from t5x, but for now the above is good enough.

**Comment by patrickvonplaten** [comment]
> Thanks for adding this! I'm curious why the empty module? Does one of the script complain if we don't add it?

Copied it more or less from dialogpt after I noticed one check repo test was failing. Re-iterated and it looks like the only thing that is required is that the name is in `configuration_auto.py` - thanks for double-checking here!

**Pull request merged**